openssl/src/x509/mod.rs

```diff
@@ -1431,6 +1431,26 @@ impl X509ReqBuilder {
         }
     }
 
+    /// Adds attribute defined by name and type to certificate request
+    pub fn add_attribute(
+        &mut self,
+        attrname: &str,
+        attrtype: Asn1Type,
+        value: &[u8],
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            let attrname = CString::new(attrname).unwrap();
+            cvt(ffi::X509_REQ_add1_attr_by_txt(
```

Comment:
I think it'd be preferrable to have our API built around `X509_REQ_add1_attr_by_OBJ` (and take an OID as an argument)